pandas/tests/io/parser/test_python_parser_only.py

```diff
@@ -488,3 +489,77 @@ def test_header_int_do_not_infer_multiindex_names_on_different_line(python_parse
     )
     expected = DataFrame({"a": ["a", "c", "f"]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_no_thousand_convert_for_non_numeric_cols(python_parser_only):
```

Comment:
Is the difference between this test and the one below just that `thousands` is `,` instead of `.`?

I'm not really sure I see what this one adds - am I missing something, or is the one below (`test_no_thousand_with_dot_convert_for_non_numeric_cols`) enough?
I'd suggest either:
- just keeping the test below
- if checking `,` instead of `.` goes down a different path or does something different, then parametrising the test over `thousands`

Reply:
I would say that there is a difference between the `test_no_thousand_convert_for_non_numeric_cols` and the one below.
In case we have `float `as a dtype we can’t use `.` as thousands. For the example with `{"a": str, "b": np.float64, "c": np.int64}` I used `, `as thousands.

I combined two tests into one by adding additional parameters in parametrization.